vignettes/adcoeq.Rmd

```diff
@@ -0,0 +1,215 @@
+```{r echo=FALSE}
+dataset_vignette(
+  arrange(adcoeq2, USUBJID, PARCAT1, ADY, PARAMCD),
+  display_vars = exprs(USUBJID, PARAMCD, PARAM, PARCAT1, QSSTRESN, ADY, AVISIT, AVALC, AVAL),
+  filter = PARAMCD %in% c("COEQ01", "COEQ02", "COEQ03", "COEQ04", "COEQ05", "COEQ06", "COEQ07", "COEQ08", "COEQ09", "COEQ20")
+)
+```
+
+We handle unscheduled visits as normal visits. For deriving visits based on
+time-windows, see `{admiral}` [Visit and Period Variables](https://pharmaverse.github.io/admiral/articles/visits_periods.html).
+
+
+## Derive the four Subscales {#subscales}
+
+Scales and Scores are often derived as the sum or the average across a subset of
+the items. 
+For the Control of Eating Questionnaire, four subscales are derived:
+
+* Craving Control: Calculate average of items 9, 10, 11, 12 and 19.
+
+* Craving for Sweet: Calculate average of items 3, 13, 14 and 15.
+
+* Craving for Savoury: Calculate average of items 4, 16, 17 and 18.
+
+* Positive Mood: Calculate average of items 5, 7, 8 and 6 (reversed).
+
+These parameters can be derived by `derive_summary_records()`:
+```{r eval=TRUE}
+adcoeq3 <- adcoeq2 %>%
+  call_derivation(
+    derivation = derive_summary_records,
+    variable_params = list(
+      params(
+        filter_add = PARAMCD %in% c('COEQ09','COEQ10','COEQ11','COEQ12','COEQ19'),
+        set_values_to = exprs(
+          AVAL = mean(AVAL, na.rm = TRUE),
+          PARAMCD = "COEQCRCO",
+          PARAM = "COEQ - Craving Control"
+        )
+      ),
+      params(
+        filter_add = PARAMCD %in% c('COEQ03','COEQ13','COEQ14','COEQ15'),
+        set_values_to = exprs(
+          AVAL = mean(AVAL, na.rm = TRUE),
+          PARAMCD = "COEQCRSW",
+          PARAM = "COEQ - Craving for Sweet"
+        )
+      ),
+      params(
+        filter_add = PARAMCD %in% c('COEQ04','COEQ16','COEQ17','COEQ18'),
+        set_values_to = exprs(
+          AVAL = mean(AVAL, na.rm = TRUE),
+          PARAMCD = "COEQCRSA",
+          PARAM = "COEQ - Craving for Savoury"
+        )
+      ),      
+      params(
+        filter_add = PARAMCD %in% c('COEQ05','COEQ07','COEQ08','COEQ06'),
+        set_values_to = exprs(
+          AVAL = mean(AVAL, na.rm = TRUE),
+          PARAMCD = "COEQPOMO",
+          PARAM = "COEQ - Positive Mood"
+        )
+      )
+    ),
+    dataset_add = adcoeq2,
+    by_vars = exprs(STUDYID, USUBJID, AVISIT, AVISITN, ADT, ADY, PARCAT1, TRTSDT, TRTEDT, TRT01P, TRT01A)
+  )
+```
+
+```{r echo=FALSE}
+dataset_vignette(
+  arrange(adcoeq3, USUBJID, ADY, PARAMCD),
+  display_vars = exprs(USUBJID, PARAMCD, PARAM, AVAL, ADY, AVISIT),
+  filter = PARAMCD %in% c("COEQCRCO", "COEQCRSW", "COEQCRSA", "COEQPOMO")
+)
+```
+
+## Remaining ADCOEQ Set-up {#adcoeq_end}
+
+The `{admiral}` [Creating Questionnaire ADaMs vignette](https://pharmaverse.github.io/admiral/articles/questionnaires.html) 
+describes further steps like how to calculate the change from baseline variables or how to add parameters for questionnaire completion.
+
```

Comment:
Should we not add the adcoeq template link here like advs vignette ?

Reply:
Yes please add, as event if the link ends up being broken, we know it will be live in 2 weeks time.

Reply:
Done